src/logging.rs

```diff
@@ -70,14 +78,56 @@ impl tracing::field::Visit for CaptureLayerVisitor<'_> {
     }
 }
 
-pub fn plugin(app: &mut App) {
-    app.add_event::<LogEvent>();
-    app.add_systems(Update, transfer_log_events);
+pub struct LogViewerPlugin {
+    auto_open_threshold: LevelFilter,
+}
+
+impl Default for LogViewerPlugin {
+    fn default() -> Self {
+        Self {
+            auto_open_threshold: LevelFilter::OFF,
+        }
+    }
+}
 
-    app.observe(handle_log_viewer_visibilty);
+impl LogViewerPlugin {
+    pub fn new() -> Self {
```

Comment:
for now if we only have `default` lets not have `new` it indicates we do something different than `default`

Reply:
I disagree; it's considered idiomatic, and the general Rust community, as far as I'm aware, consider them synonymous, to the point of there being a default warning lint for saying you should implement `Default` if `pub fn new() -> Self` exists, since they're presumed the same.